pkg/pmem-csi-operator/controller/deployment/deployment_controller.go

```diff
@@ -128,7 +133,14 @@ func (r *ReconcileDeployment) Reconcile(request reconcile.Request) (reconcile.Re
 		}
 	}()
 
-	if err := deployment.EnsureDefaults(); err != nil {
+	driverImage, err := r.ContainerImage()
+	if err != nil {
+		klog.Warningf("Failed to find the operator image: %v", err)
```

Comment:
If we want to treat this as error then, we should move this call to the operator initialization phase, instead of reconciling loop. Ok, i will do the needed changes.

Reply:
Done, As per now new implementation, the operator exists if fail to get self-image.